workspacemanager/WorkspaceManagerPresenterTest.py

```diff
@@ -0,0 +1,173 @@
+from mock.mock import self
+
+from WorkspaceManagerPresenter import WorkspaceManagerPresenter
+from WorkspaceProvider import WorkspaceProvider
+from WorkspaceView import WorkspaceView
+from command import Command
+import unittest
+import mock
+from mock import call
+
+
+#ToDo askOwen, view related tests, i.e view returns 'selected_workspace' that does not exist
+#TODO handle mantid load fail inquiry
+#TODO Test constructor and make this test specific
+
+class WorkspaceManagerPresenterTest(unittest.TestCase):
+    def setUp(self):
+        self.workspace_provider = mock.create_autospec(spec=WorkspaceProvider)
+        self.view = mock.create_autospec(spec=WorkspaceView)
+
+
+
+    def test_load_one_workspace(self):
+        self.presenter = WorkspaceManagerPresenter(self.view, self.workspace_provider)
+        self.view.get_workspace_to_load_path = mock.Mock(return_value=r'C:\a\b\cde.nxs')
+        self.workspace_provider.getWorkspaceNames = mock.Mock(return_value=['cde'])
+        self.presenter.notify(Command.LoadWorkspace)
+        self.view.get_workspace_to_load_path.assert_called_once()
+        self.workspace_provider.Load.assert_called_with(Filename=r'C:\a\b\cde.nxs', OutputWorkspace='cde')
+        self.view.display_loaded_workspaces.assert_called_with(['cde']) #TODO could this be has is_called?
+
+    def test_load_multiple_workspaces(self):
+        self.presenter = WorkspaceManagerPresenter(self.view, self.workspace_provider)
+        self.view.get_workspace_to_load_path = mock.Mock(
+            side_effect=[r'C:\path\to\file1.nxs', r'C:\path\to\any\file2.nxs', r'C:\path\to\file3.nxs'])
+        self.workspace_provider.getWorkspaceNames = mock.Mock(side_effect=[['file1'], ['file1', 'file2'],
+                                                                           ['file1', 'file2', 'file3']])
+        for i in range(3):
+            self.presenter.notify(Command.LoadWorkspace)
+        load_calls = [call(Filename='C:\\path\\to\\file1.nxs', OutputWorkspace='file1'),
+                      call(Filename='C:\\path\\to\\any\\file2.nxs', OutputWorkspace='file2'),
+                      call(Filename='C:\\path\\to\\file3.nxs', OutputWorkspace='file3')]
+        self.workspace_provider.Load.assert_has_calls(load_calls)
+        #The test should just finish here
+        # <Not Maybe not useful> TODO askOwen case1
```

Comment:
Either possibility is fine that you mention here. Lets keep the full option (verbose) for now.